rpi_backlight.py

```diff
@@ -38,97 +39,174 @@ def _get_value(name: str) -> str:
 
 def _set_value(name: str, value: Any) -> None:
     try:
-        with open(os.path.join(PATH, name), "w") as f:
+        with open(os.path.join(path, name), "w") as f:
             f.write(str(value))
     except (OSError, IOError) as err:
         if err.errno == 13:
             _perm_denied()
 
 
-def get_actual_brightness() -> int:
-    """Return the actual display brightness."""
-    return int(_get_value("actual_brightness"))
+def set_brightness_value(value: Any) -> None:
+    if mode == "MODE_TINKERBOARD":
+        _set_value("tinker_mcu_bl", value)
+    elif mode == "MODE_RPI":
+        _set_value("bl_power", value)
 
+def get_brightness_value() -> int:
+    """Return the actual display brightness."""
+    if mode == "MODE_TINKERBOARD":
+        return int(_get_value("tinker_mcu_bl"))
+    elif mode == "MODE_RPI":
+        return int(_get_value("actual_brightness"))
 
 def get_max_brightness() -> int:
     """Return the maximum display brightness."""
-    return int(_get_value("max_brightness"))
-
+    if mode == "MODE_TINKERBOARD":
+        return 255
+    elif mode == "MODE_RPI":
+        return int(_get_value("max_brightness"))
+    
 
 def get_power() -> bool:
-    """Return wether the display is powered on or not."""
+    """Return whether the display is powered on or not."""
     # 0 is on, 1 is off
-    return not int(_get_value("bl_power"))
+    if mode == "MODE_TINKERBOARD":
+        return not int(_get_value("tinker_mcu_bl"))
+    elif mode == "MODE_RPI":
+        return not int(_get_value("bl_power"))
 
 
-def set_brightness(value: int, smooth: bool = False, duration: float = 1) -> None:
+def set_brightness(value: int, smooth: bool = True, duration: float = 1) -> None:
     """Set the display brightness.
 
-    :param value: Brightness value between 11 and 255
+    :param value: Brightness value between 0 and 255
     :param smooth: Boolean if the brightness should be faded or not
     :param duration: Fading duration in seconds
     """
-
     max_value = get_max_brightness()
     if not isinstance(value, int):
         raise ValueError("integer required, got '{}'".format(type(value)))
-    if not 10 < value <= max_value:
-        raise ValueError(
-            "value must be between 11 and {}, got {}".format(max_value, value)
-        )
+
+    if mode == "MODE_TINKERBOARD":
+            if not -1 < value <= max_value:
+                raise ValueError(
+                    "value must be between 0 and {}, got {}".format(max_value, value)
+                )
+    elif mode == "MODE_RPI":
+        if not -10 < value <= max_value:
```

Comment:
I'm ok with changing the min value to 0, not 11, as far as I can remember the display would turn off completely for values of 10 or below. I might be wrong here. But if you change this, please update the documentation as well and remove this block, it's duplicated code.

Reply:
fixed